src/main/java/com/josketres/rfcfacil/VerificationDigitCalculator.java

```diff
@@ -69,10 +69,14 @@ public String calculate() {
         int reminder = sum % 11;
         if (reminder == 0) {
             return "0";
-        } else if (reminder == 10) {
-            return "A";
         } else {
-            return String.valueOf(11 - reminder);
+            int result = 11 - reminder;
+
+            if (result == 10) {
+                return "A";
+            } else {
+                return String.valueOf(11 - reminder);
```

Comment:
Estoy tratando de hacer unas modificaciones más en la generación de los RFC
para personas morales, he probado varios casos pero me arrojan Homoclaves y
DV diferentes, ya identifiqué algunos casos, mas tarde actualizo para poder
revisar.

Saludos.

ISC. Mario Jiménez Carrasco.
Subdirección Técnico y de Negocios
Desarrollo de Software.
Gobierna SCP.

On Wed, Jun 29, 2016 at 2:28 AM, Josue Zarzosa <notifications@github.com>
wrote:

> In src/main/java/com/josketres/rfcfacil/VerificationDigitCalculator.java
> <https://github.com/josketres/rfc-facil/pull/4#discussion_r68897232>:
>
> >          } else {
> > -            return String.valueOf(11 - reminder);
> > +            int result = 11 - reminder;
> > +
> > +            if (result == 10) {
> > +                return "A";
> > +            } else {
> > +                return String.valueOf(11 - reminder);
>
> Después de integrar esta corrección la cambié un poquito.
> Esta es la versión final (creo que es más elegante):
>
>     public String calculate() {
>
>         int sum = 0;
>         for (int i = 0; i < 12; i++) {
>             sum += mapDigit(rfc12Digits.charAt(i)) * (13 - i);
>         }
>
>         int reminder = sum % 11;
>         if (reminder == 0) {
>             return "0";
>         } else {
>             return Integer.toHexString(11 - reminder).toUpperCase(); // from 1 to A (hex)
>         }
>     }
>
> Puedes ver la version actual aqui
> <https://github.com/josketres/rfc-facil/blob/master/src/main/java/com/josketres/rfcfacil/VerificationDigitCalculator.java#L62>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/josketres/rfc-facil/pull/4/files/a939e9b5832c6ee4823fed964dcaeae51ffb4c76#r68897232>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AE4qFdvnJ1FN1IrxKVgptNW7M5RgfBFJks5qQh6cgaJpZM4I_k1w>
> .
>